### PR TITLE
Check if environment is production-like in stripped-build-plugins

### PR DESCRIPTION
**Pull request by IceDragon200**
On request, a PR, to fix the production like environment issue

## Related

#5233

### CODE REVIEW

**Comment by rwjblue** [comment]
Seems good for us to be consistent here, but there are only three possible values for `process.env.EMBER_ENV` that are valid: `test`, `development`, and `production`.

I'd very slightly prefer to change the `index.js` check to be `=== 'production'` (instead of changing the other to use a regexp)...

**Comment by runspired** [comment]
@rwjblue I agree that ember-cli doesn't support it in theory but I also know that historically we've loosely matched `production` and many teams have relied on that for configuring production-like staging environments (often began before ember-cli-deploy paved the path for nice things).

**Comment by IceDragon200** [comment]
@rwjblue, @runspired Well prior to updating to 2.16 (coming from 2.9)  it was working fine with the production-like environments, it kinda threw a wrench in my day after the upgrade.

Maybe keeping it loose for now but adding a deprecation warning if a production-like environment is found:

```javascript
if (env !== 'production' && /production/.test(env)) {
 warn("production like environments are deprecated, if using ember-cli-deploy, please configure your build using 'production'");
}
```

This would ensure folks like me, who expected the old behaviour are given fair warning and a path for migration before completely flipping the table on them.

Sorry about the late reply, it was a tricky week

**Comment by bmac** [comment]
@IceDragon200 that seems like a reasonable compromise. Do you need this change back-ported into 2.16?

**Comment by IceDragon200** [comment]
@bmac Well I've already bitten the bullet and changed my code to use environment variables, so, not for me, but for anyone else that encounters this problem.

tl;dr Sure go ahead :)

**Comment by runspired** [comment]
@IceDragon200 fwiw what I was saying was that as long as both checks match this should resolve your issue. Granted `===` means you won't get the code stripping in production-like environments, but it won't error either.  I agree with @bmac that a deprecation for production-like is reasonable.

**Comment by bmac** [comment]
I added a pr for introducing the production like warning. https://github.com/emberjs/data/pull/5239
